src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -467,6 +467,7 @@ void Unit::Update(uint32 p_time)
 
     UpdateSplineMovement(p_time);
     i_motionMaster->UpdateMotion(p_time);
+    UpdateUnderwaterState(GetMap(), GetPositionX(), GetPositionY(), GetPositionZ());
```

Comment:
This method was called in UpdateMotion() (the line above). So it was called on each Unit::Update() call anyway. And putting this call here is better than in MotionMaster::UpdateMotion() imo.